src/stan/math/prim/scal/meta/get.hpp

```diff
@@ -6,8 +6,7 @@
 
 namespace stan {
 
-  template <typename T>
-  inline T get(const T& x, size_t n) {
+  inline double get(double x, size_t n) {
```

Comment:
Yeah, I made similar changes to your other meta-programming functions. Again, the problem is that `template<typename T>` matches `Eigen::MatrixBase<T>` even when there is a more specific template specification for `Eigen::MatrixBase`.